aip/auth/4115.md

```diff
@@ -5,124 +5,110 @@ state: approved
+the remote API’s mTLS endpoint is out of the scope of this AIP.
 
-To detect if the application is running on Compute Engine or an equivalent runtime,
-the auth library **should** depend on the [Metadata Service Library][4].
+## Guidance
 
-### Compute Engine or Equivalent Runtime
+### Access Default mTLS Credentials
 
-If the application runs on a Compute Engine instance (or equivalent runtime),
-the auth library can request the following URL to get an access token associated
-with instance:
+To leverage Google’s default mTLS credentials, the client **should** retrieve configurations from 
+MDS. All virtual environments (GCE, GAE and GKE) MDS expose a HTTP endpoint (mTLS configuration 
+endpoint) that serves mTLS configurations.
 
+The MDS mTLS configuration endpoint is: 
 ```
-http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token
+http://169.254.169.254/computeMetadata/v1/googleAutoMtlsConfiguration
 ```
 
-In response, the auth library will usually receive a token in the following JSON
-format:
+The request to the MDS endpoint **should** be a HTTP GET request without any parameter or payload.
 
-```json
-{
-      "access_token":"YOUR_ACCESS_TOKEN",
-      "expires_in":3599,
-      "token_type":"Bearer"
- }
-```
+The response from the MDS endpoint **should** contain the following information:
 
-#### Scopes
-Access tokens obtained from the metadata server contain the scopes specified at instance creation.
-For example, if an instance is granted only the
-https://www.googleapis.com/auth/storage-full scope for Cloud Storage, then the tokens obtained 
-from the instance's metadata server have only the `storage-full` scope.
+* The **Secure Session Agent** address: the client doesn’t have direct access to mTLS credentials. 
+  The Secure Session Agent manages default mTLS credentials. The client can only use mTLS 
+  credentials through Secure Session Agent. The address can be an IP:port address or a Unix Domain 
+  Socket (UDS).
 
-When running on GCE, the metadata server ignores requests for custom scopes.
-On GCE equivalent runtimes, clients can request a different set of scopes to the metadata server
-using the `scopes` url parameter, e.g.:
+The client **must** follow the steps below to access Google’s default mTLS credentials.
 
-```
-http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes=comma-separated-list-of-scopes
-```
-
-The auth library **should** allow the caller to optionally specify a list of custom scopes,
-and add the `scopes` parameter to the request when needed.
-Depending on the runtime environment, the request for custom scopes will be transparently
-ignored by the server (GCE) or fulfilled (GCE equivalent).
+1. Check if the remote endpoint supports mTLS. 
+   * If yes, go to step (2).
+   * If not, go to step (3). 
+2. Send a request to the MDS mTLS configuration endpoint. If the request is successful and the 
```

Comment:
nit: Same here, it should either be "MDS's mTLS configuration endpoint" or "mTLS configuration endpoint". And please be consistent with whatever we chose above.

Reply:
Done.